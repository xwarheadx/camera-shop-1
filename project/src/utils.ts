import { PAGE_LIMIT } from './consts';
import { reviewDataType } from '../src/types/review';

export function getPagesCount(cameraCount: number): number {
  return Math.ceil(cameraCount / PAGE_LIMIT);
}

export function getNumeric(page: string) {
  return Number(page.replace( /^\D+/g, ''));
}

export function seperatePrice(priceValue: number) {
  let returnedPrice = String(priceValue);
  if (returnedPrice.length > 3) {
    returnedPrice = `${returnedPrice.slice(0,2)} ${returnedPrice.slice(2)}`;
  }
  return returnedPrice;
}

export function validateInput(target: (EventTarget & HTMLInputElement) | (EventTarget & HTMLTextAreaElement)) {
  if (target.parentElement?.parentElement && target.name === 'rating' && (Number(target.value) > 0)) {
    if (!target.parentElement.parentElement.parentElement?.className.includes('is-valid')) {
      target.parentElement.parentElement.parentElement?.classList.add('is-valid');
      target.parentElement.parentElement.parentElement?.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name !== 'review' && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length > 0) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string'){
    if(target.value.length === 0) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

  if (target.parentElement?.parentElement && typeof(target.value) === 'string') {
    if (!target.parentElement.parentElement.className.includes('is-valid') && target.value.length >= 5) {
      target.parentElement.parentElement.className += ' is-valid';
      target.parentElement.parentElement.classList.remove('is-invalid');
    }
  }

  if (target.parentElement?.parentElement && target.name === 'review' && target.value.length < 5) {
    if (!target.parentElement.parentElement.className.includes('is-invalid')) {
      target.parentElement.parentElement.className += ' is-invalid';
      target.parentElement.parentElement.classList.remove('is-valid');
    }
  }

}
export function validateForm(data: reviewDataType) {
  let validationStatus = true;
  Object.entries(data).forEach(([key, value]) =>{
    if (key !== 'review') {
      if(value === '') {
        const failedInput = document.querySelector(`input[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
        validationStatus = false;
      }
    }
    if(key === 'review') {
      if(String(value).length < 5) {
        const failedInput = document.querySelector(`textarea[name=${key}]`);
        validateInput(failedInput as HTMLInputElement);
      }
    }
    if(key === 'rating') {
      if(value < 1) {
        const failedInput = document.querySelector(`input[name=${key}]`);
        if(failedInput && failedInput.parentElement?.parentElement){
          failedInput.parentElement.parentElement.parentElement?.classList.add('is-invalid');
        }
        validationStatus = false;
      }
    }
  });
  return validationStatus;
}


