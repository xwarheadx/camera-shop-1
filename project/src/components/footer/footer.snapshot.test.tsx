import { renderWithProviders } from '../../test/utils/render-with-redux';
import Footer from './footer';

test('Render Footer', () => {
  const { asFragment } = renderWithProviders(<Footer/>);
  expect(asFragment()).toMatchSnapshot();
});
