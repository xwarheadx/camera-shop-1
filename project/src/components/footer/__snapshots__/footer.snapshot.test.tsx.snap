// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Footer 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div
      class="container"
    >
      <div
        class="footer__info"
      >
        <a
          aria-label="Переход на главную"
          class="footer__logo"
          href="/"
        >
          <svg
            aria-hidden="true"
            height="36"
            width="100"
          >
            <use
              xlink:href="#icon-logo-mono"
            />
          </svg>
        </a>
        <p
          class="footer__description"
        >
          Интернет-магазин фото- и видеотехники
        </p>
        <ul
          class="social"
        >
          <li
            class="social__item"
          >
            <a
              aria-label="Переход на страницу вконтатке"
              class="link"
              data-testid="logo-test"
              href="/"
            >
              <svg
                aria-hidden="true"
                height="20"
                width="20"
              >
                <use
                  xlink:href="#icon-vk"
                />
              </svg>
            </a>
          </li>
          <li
            class="social__item"
          >
            <a
              aria-label="Переход на страницу pinterest"
              class="link"
              href="/"
            >
              <svg
                aria-hidden="true"
                height="20"
                width="20"
              >
                <use
                  xlink:href="#icon-pinterest"
                />
              </svg>
            </a>
          </li>
          <li
            class="social__item"
          >
            <a
              aria-label="Переход на страницу reddit"
              class="link"
              href="/"
            >
              <svg
                aria-hidden="true"
                height="20"
                width="20"
              >
                <use
                  xlink:href="#icon-reddit"
                />
              </svg>
            </a>
          </li>
        </ul>
      </div>
      <ul
        class="footer__nav"
      >
        <li
          class="footer__nav-item"
        >
          <p
            class="footer__title"
          >
            Навигация
          </p>
          <ul
            class="footer__list"
          >
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="/catalog"
              >
                Каталог
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="/"
              >
                Гарантии
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="/"
              >
                Доставка
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="/"
              >
                О компании
              </a>
            </li>
          </ul>
        </li>
        <li
          class="footer__nav-item"
        >
          <p
            class="footer__title"
          >
            Ресурсы
          </p>
          <ul
            class="footer__list"
          >
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="/"
              >
                Курсы операторов
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="/"
              >
                Блог
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="/"
              >
                Сообщество
              </a>
            </li>
          </ul>
        </li>
        <li
          class="footer__nav-item"
        >
          <p
            class="footer__title"
          >
            Поддержка
          </p>
          <ul
            class="footer__list"
          >
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="/"
              >
                FAQ
              </a>
            </li>
            <li
              class="footer__item"
            >
              <a
                class="link"
                href="/"
              >
                Задать вопрос
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </footer>
</DocumentFragment>
`;
